modules/reindex/src/yamlRestTest/resources/rest-api-spec/test/update_by_query/90_tsdb.yml

```diff
@@ -77,7 +77,7 @@ update tag field:
 update dimension field:
   # TODO better error message
   - do:
-      catch: bad_request
+      catch: conflict
```

Comment:
I don't fully follow why the exception type changed here. Since the original exception is still thrown in TsidExtractingIdFieldMapper.java line 74. I would expect no change here.

Reply:
The error is

```
"type" : "version_conflict_engine_exception",
            "reason" : "[lkPfLx162czh7Xh9AAABeRnRFAY]: version conflict, required seqNo [0], primary term [1]. but no document was found",
```

The logic you mention is for bwc, not exercised in the new code path. I'm not very familiar with painless, we can do something different than dummy routing if preferred.

Reply:
Ok, I think the changes to this yaml file can be undone by moving:

```
if (context.sourceToParse().id() != null && false == context.sourceToParse().id().equals(id)) {
            throw new IllegalArgumentException(
                String.format(
                    Locale.ROOT,
                    "_id must be unset or set to [%s] but was [%s] because [%s] is in time_series mode",
                    id,
                    context.sourceToParse().id(),
                    context.indexSettings().getIndexMetadata().getIndex().getName()
                )
            );
        }
```

To before:

```
context.id(id);
```

In `TsidExtractingIdFieldMapper#createField(...)` method. I think this if statement shouldn't be part of just the bwc logic, but remain the be active in general.

Reply:
I had that before, it fails when reindexing I think.

Reply:
I ran `./gradlew -p modules/reindex yamlRestTest` here locally and that did pass with this change.

Reply:
Oh cool, your suggestion to get routing from DocumentParserCoxtext probably did the trick for this too.